app/controllers/logs_controller.rb

```diff
@@ -301,19 +301,10 @@ def leave
   end
 
   def receipt
-    if Date.valid_date?(params[:start_date][:year].to_i, params[:start_date][:month].to_i, params[:start_date][:day].to_i)
-      @start_date = Date.new(params[:start_date][:year].to_i, params[:start_date][:month].to_i, params[:start_date][:day].to_i)
-    else
-      flash[:notice] = 'Invalid Date Set for Start Date. Please try again!'
-      return redirect_to(request.referer || root_path)
-    end
+    parse_start_stop_date(params)
 
-    if Date.valid_date?(params[:stop_date][:year].to_i, params[:stop_date][:month].to_i, params[:stop_date][:day].to_i)
-      @stop_date = Date.new(params[:stop_date][:year].to_i, params[:stop_date][:month].to_i, params[:stop_date][:day].to_i)
-    else
-      flash[:notice] = 'Invalid Date Set for End Date. Please try again!'
-      return redirect_to(request.referer || root_path)
-    end
+    @start_date = Date.new(params[:start_date][:year].to_i, params[:start_date][:month].to_i, params[:start_date][:day].to_i)
+    @stop_date = Date.new(params[:stop_date][:year].to_i, params[:stop_date][:month].to_i, params[:stop_date][:day].to_i)
```

Comment:
This is another option to try to clean things up here: 

`@start_date, @stop_date = parse_start_stop_dates`

And in `parse_start_stop_dates` you could do the `Date.new(params[:start_date][:year]` and `Date.new(params[:stop_date][:year].to_i` and return as an array. 

OR you can just set `@start_date` and `@stop_date` in the `parse_start_stop_dates` method as that's a valid way of doing this in Rails. I think this is the best option.

In that case i'd name it `parse_and_set_dates` or `parse_and_set_start_stop_dates` if you wanted to be really clear